crates/astria-core/src/sequencerblock/v1/block.rs

```diff
@@ -783,21 +783,21 @@ impl SequencerBlock {
     ) -> Result<Self, SequencerBlockError> {
         use prost::Message as _;
 
-        let tree = merkle_tree_from_data(&data);
-        let data_hash = tree.root();
-
-        let mut data_list = data.into_iter();
-
-        // TODO: this needs to go into the block header
-        let _extended_commit_info = data_list
-            .next()
-            .ok_or(SequencerBlockError::no_extended_commit_info())?;
-
-        let (rollup_transactions_root, rollup_ids_root) =
-            rollup_transactions_and_ids_root_from_data(&mut data_list)?;
+        let InitialDataElements {
+            data_root_hash,
+            // This iterator over `data` has been advanced by 3 steps while retrieving the initial
+            // elements. The next element to be yielded will be the first of the actual rollup txs.
+            data_iter,
+            // TODO: this needs to go into the block header
+            _extended_commit_info,
+            rollup_transactions_root,
+            rollup_transactions_proof,
+            rollup_ids_root,
+            rollup_ids_proof,
+        } = take_initial_elements_from_data(data)?;
```

Comment:
note to self: need to allow this to not have the extended_commit_info as to be nonbreaking